tests/unit/tracing_test.cpp

```diff
@@ -0,0 +1,61 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include "mocking.h"
+
+using namespace testing;
+
+extern "C" {
+   #include <tilck/kernel/test/tracing.h>
+}
+
+
+class tracing_mock : public KernelSingleton {
+public:
+   MOCK_METHOD(int, handle_sys_trace_arg, (const char *arg), (override));
+};
+
+TEST(simple_wildcard_match, basic)
+{
+   ASSERT_TRUE(simple_wildcard_match("test string", "test string"));
+   ASSERT_TRUE(simple_wildcard_match("test string", "test string*"));
+   ASSERT_TRUE(simple_wildcard_match("test string", "tes? string"));
+   ASSERT_TRUE(simple_wildcard_match("test string", "test *"));
+   ASSERT_FALSE(simple_wildcard_match("test string", "test strin1"));
+   ASSERT_FALSE(simple_wildcard_match("test string", "te*t"));
+   ASSERT_FALSE(simple_wildcard_match("test string", "test string1"));
+}
+
+TEST(set_traced_syscalls_int, basic)
+{
+   tracing_mock mock;
+   const char too_long[TRACED_SYSCALLS_STR_LEN] = {'a'};
```

Comment:
1. Don't initialize this to with `{'a'}` as it's doesn't make any sense. You're using memset already.
2. You are not zero-terminating the string after memset.
3. You don't really need two arrays like that. It's enough to have just one and calling it `tracing_str` and then make it shorter by moving the zero-termination.
4. Please, learn to use vertical space. 
5. Separate variable declarations from memset calls. First the array and the other variables, then vertical space, then the memset calls and the other stuff.